server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java

```diff
@@ -87,6 +87,9 @@ public BulkRequest(StreamInput in) throws IOException {
         requests.addAll(in.readCollectionAsList(i -> DocWriteRequest.readDocumentRequest(null, i)));
         refreshPolicy = RefreshPolicy.readFrom(in);
         timeout = in.readTimeValue();
+        for (DocWriteRequest<?> request : requests) {
+            indices.add(request.index());
```

Comment:
Can we assert `request.index()` is not `null`? I think it can be `null` when [constructing](https://github.com/elastic/elasticsearch/blob/c37fa227bd2ea2f16e013daa11d85bdf3c538d4e/server/src/main/java/org/elasticsearch/action/bulk/BulkRequest.java#L150) the bulk request initially. But it should not be `null` once it is sent across the wire.

Reply:
ok, let me take a look at that.

Reply:
I added the assert Yang. Please review.

Reply:
Nit: We can inline `Objects.requireNonNull(...)` call into `indices.add` since it returns the argument.

Reply:
```
indices.add(Objects.requireNonNull(request.index(), "'request index' must not be null"))
```

Reply:
Nice!